### DISCUSSION TITLE
HAL (or audio codec driver library) for AI Thinker audiokit with es8388

### DISCUSSION THREAD

**Discussion by dkotte** [Edzelf/ESP32Radio-V2#117] [General]
Hello everyone,
I'm trying to get an AI Thinker audiokit with es8388 to output audio.
Unfortunately, the programmer of the AC101 audio codec driver library hasn't written this for the es8388 (until now).
(compare https://github.com/yveaux/AC101)
But he refers to some sources used, which wrote ES8388 interfaces for also.
Is there someone who understands something like this and can adapt the "audio codec driver library" so initialization and control commands will work?

More details will follow

**Comment by Edzelf** [comment]
There is a driver [here](https://github.com/espressif/esp-adf/blob/master/components/audio_hal/driver/es8388/es8388.c).
